examples/basic/index.html

```diff
@@ -31,29 +34,38 @@
     client.searchForInstances(searchInstanceOptions).then((instances) => {
       const promises = []
       for (let i = 0; i < instances.length; i++) {
+        if (instances[i] === undefined ||
+          instances[i]["0020000E"] === undefined ||
+          instances[i]["00080018"] === undefined) {
+          continue;
+        }
         const sopInstanceUID = instances[i]["00080018"]["Value"][0];
         const retrieveInstanceOptions = {
           studyInstanceUID,
           seriesInstanceUID,
           sopInstanceUID,
         };
+
         const promise = client.retrieveInstanceMetadata(retrieveInstanceOptions).then(metadata => {
-          const imageType = metadata[0]["00080008"]["Value"];
-          if (imageType[2] === "VOLUME") {
-            return(metadata[0]);
+          const imageType = metadata[0]["00080008"]
```

Comment:
As a rule I dislike code that uses dicom hex for any tags that are defined in the dictionary.  It makes the code less readable and is partly why dicom has such a bad reputation.  Can we change these?

Reply:
Fully agree with @pieper. That's why I am suggesting to immediately format (sanitize) the metadata.

Reply:
In dcmjs, the pattern is to define a [subclass of Normalizer](https://github.com/dcmjs-org/dcmjs/blob/master/src/normalizers.js#L33-L57) for each SOPClass to catch and correct known deviations from the standard.  This can avoid spreading special case code all over the application logic.